OngProject/Controllers/CategoriesController.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using OngProject.Core.Interfaces;
+using OngProject.DataAccess.UnitOfWork.Interfaces;
+using OngProject.Entities;
+using System;
+using System.Threading.Tasks;
+using AutoMapper;
+using OngProject.Core.Models;
+
+namespace OngProject.Controllers
+{
+    [ApiController]
+    public class CategoriesController : ControllerBase
+    {
+        private readonly IUnitOfWork _uow;
+        private readonly ICategoriesBusiness _categoriesBusiness;
+        private readonly IMapper _mapper;
+        public CategoriesController(IUnitOfWork uow, ICategoriesBusiness categoriesBusiness, IMapper mapper)
+        {
+            this._uow = uow;
+            this._categoriesBusiness = categoriesBusiness;
+            this._mapper = mapper;
+        }
+
+        // GET List/Categories
+        [HttpGet]    
+        [Route("List/Categories")]
+        public async Task<IActionResult> GetAllCategories() 
+        {
+            try
+            {
+                var categories =  _categoriesBusiness.FindCategoryAsync(m => m.softDelete != true);
+                return categories != null ? Ok(_mapper.Map<IEnumerable<CategoriesDTO>>(categories)) 
+                                       : NotFound("The list of categories has not been found");                
+            }
+            catch (System.Exception ex)
+            {
+                throw new Exception(ex.Message);
+            }           
+        }
+
+        // GET List/CategoriesById
+        [HttpGet]        
+        [Route("List/CategoriesById/{id}")]
+        public async Task<IActionResult> GetById(int id)
+        {
+            try
+            {
+                if(id == 0)
+                    return NotFound("Please, set an ID.");
+
+                var member = await _categoriesBusiness.GetCategoryByIdAsync(id);
+                return member != null ? Ok(_mapper.Map<CategoriesDTO>(member)) 
+                                      : NotFound("Member doesn't exists");            
+            }
+            catch (System.Exception ex)
+            {
+                throw new Exception(ex.Message);
+            }
+        }
+
+        // POST Create/Category
+        [HttpPost]       
+        [Route("Create/Category")]
+        public async Task<IActionResult> Create([FromBody] CategoriesDTO model)
+        {          
+            if(ModelState.IsValid)
+            {
+                try
+                {
+                    // validations                    
+                    if(string.IsNullOrEmpty(model.Name))
+                    {
+                        return Ok("Name required");                    
```

Comment:
Cambiaria el Ok por un status code 400, ya que interpreto que estas informando que un campo es requerido. Con un status code 400 le haces saber al cliente que envió los datos que ejecución del recurso no fue exitosa. Por otra parte, si usas DataAnnotations, que son las anotaciones [Required] en el CategoriesDTO por ejemplo para el campo Name seria:
[Required(ErrorMessage = "Name is required")]
public string Name { get; set; }
Al hacer usar esa anotacion, no es necesario usar el string.IsNullOrEmpty